### ISSUE TITLE
bash installer  - problem with crontab scripts

### ISSUE THREAD

**Issue by paolovisintin** [sipcapture/homer-installer#1]
when running /opt/homer_rotate i receive : 
BI connect('homer_data:localhost:3306','homer_user',...) failed: Access denied for user 'homer_user'@'localhost' (using password: YES) at /opt/homer_mysql_partrotate_unixtimestamp.pl line 57.
DBI connect('homer_data:localhost:3306','homer_user',...) failed: Access denied for user 'homer_user'@'localhost' (using password: YES) at /opt/homer_mysql_new_table.pl line 118.

Needs to be updated with new "method" as username is now "homer" and password is randomized

**Comment by lmangani** [comment]
good catch! fixing....

**Comment by lmangani** [comment]
Patched in 47eccc62b890d3f9a299b256007e9c8718e9dd31 could you check if its smooth and confirm?

**Comment by paolovisintin** [comment]
Recreating VM and installing via bash script [Debian 8], still getting errors :

```
root@homer5:~# /opt/homer_rotate
DBI connect('homer_data:localhost:3306','homer_user',...) failed: Access denied for user 'homer_user'@'localhost' (using password: YES) at /opt/homer_mysql_new_table.pl line 118.
Can't call method "do" on an undefined value at /opt/homer_mysql_new_table.pl line 189.
DBI connect('homer_data:localhost:3306','homer_user',...) failed: Access denied for user 'homer_user'@'localhost' (using password: YES) at /opt/homer_mysql_partrotate_unixtimestamp.pl line 57.
Can't call method "prepare" on an undefined value at /opt/homer_mysql_partrotate_unixtimestamp.pl line 70.
DBI connect('homer_data:localhost:3306','homer_user',...) failed: Access denied for user 'homer_user'@'localhost' (using password: YES) at /opt/homer_mysql_partrotate_unixtimestamp.pl line 57.
Can't call method "prepare" on an undefined value at /opt/homer_mysql_partrotate_unixtimestamp.pl line 70.
DBI connect('homer_data:localhost:3306','homer_user',...) failed: Access denied for user 'homer_user'@'localhost' (using password: YES) at /opt/homer_mysql_partrotate_unixtimestamp.pl line 57.
Can't call method "prepare" on an undefined value at /opt/homer_mysql_partrotate_unixtimestamp.pl line 70.
DBI connect('homer_statistic:localhost:3306','homer_user',...) failed: Access denied for user 'homer_user'@'localhost' (using password: YES) at /opt/homer_mysql_partrotate_unixtimestamp.pl line 57.
Can't call method "prepare" on an undefined value at /opt/homer_mysql_partrotate_unixtimestamp.pl line 70.
DBI connect('homer_statistic:localhost:3306','homer_user',...) failed: Access denied for user 'homer_user'@'localhost' (using password: YES) at /opt/homer_mysql_partrotate_unixtimestamp.pl line 57.
Can't call method "prepare" on an undefined value at /opt/homer_mysql_partrotate_unixtimestamp.pl line 70.
DBI connect('homer_statistic:localhost:3306','homer_user',...) failed: Access denied for user 'homer_user'@'localhost' (using password: YES) at /opt/homer_mysql_partrotate_unixtimestamp.pl line 57.
Can't call method "prepare" on an undefined value at /opt/homer_mysql_partrotate_unixtimestamp.pl line 70.
DBI connect('homer_statistic:localhost:3306','homer_user',...) failed: Access denied for user 'homer_user'@'localhost' (using password: YES) at /opt/homer_mysql_partrotate_unixtimestamp.pl line 57.
Can't call method "prepare" on an undefined value at /opt/homer_mysql_partrotate_unixtimestamp.pl line 70.
DBI connect('homer_statistic:localhost:3306','homer_user',...) failed: Access denied for user 'homer_user'@'localhost' (using password: YES) at /opt/homer_mysql_partrotate_unixtimestamp.pl line 57.
Can't call method "prepare" on an undefined value at /opt/homer_mysql_partrotate_unixtimestamp.pl line 70.
DBI connect('homer_statistic:localhost:3306','homer_user',...) failed: Access denied for user 'homer_user'@'localhost' (using password: YES) at /opt/homer_mysql_partrotate_unixtimestamp.pl line 57.
Can't call method "prepare" on an undefined value at /opt/homer_mysql_partrotate_unixtimestamp.pl line 70.
mysql: [Warning] Using a password on the command line interface can be insecure.
```

**Comment by lmangani** [comment]
Hi Paolo

could you check the credentials in the homer_rotate script and kamailio cfg and confirm whats there? Seems still homer user....

**Comment by paolovisintin** [comment]
Hi Lorenzo, 
credentials are the same! 
despite looking in homer_mysql_new_table.pl i can see hardcoded values:

```
27 $mysql_dbname = "homer_data";
 28 $mysql_user = "homer_user";
 29 $mysql_password = "homer_password";
 30 $mysql_host = "localhost";
```

**Comment by paolovisintin** [comment]
After correcting credentials I see other warnings : 

```
root@homer5:~# /opt/homer_rotate
DBD::mysql::db do failed: Invalid default value for 'to_date' at /opt/homer_mysql_partrotate_unixtimestamp.pl line 160.
Failed to execute query [Invalid default value for 'to_date'] with error: DBD::mysql::db do failed: Invalid default value for 'to_date' at /opt/homer_mysql_partrotate_unixtimestamp.pl line 160.
Failed to execute query [Invalid default value for 'to_date'] with error: DBD::mysql::db do failed: Invalid default value for 'to_date' at /opt/homer_mysql_partrotate_unixtimestamp.pl line 160.
Failed to execute query [Invalid default value for 'to_date'] with error: DBD::mysql::db do failed: Invalid default value for 'to_date' at /opt/homer_mysql_partrotate_unixtimestamp.pl line 160.
Failed to execute query [Invalid default value for 'to_date'] with error: DBD::mysql::db do failed: Invalid default value for 'to_date' at /opt/homer_mysql_partrotate_unixtimestamp.pl line 160.
```

**Comment by adubovikov** [comment]
please disable strict mode in your my.cnf

**Comment by paolovisintin** [comment]
After changing /etc/mysql/my.cnf (adding sql-mode="") ok for `Invalid default value` !

**Comment by lmangani** [comment]
two tier fix: 

patch installer and deb/rpm installers for pl file credentials (done) 
in parallel a fix on the schema input files will take place

**Comment by emdici** [comment]
Hi guy's 

I'm trying to install Homer on centos 7 with the bash script...
after a couple of installs i find out the right way to make it work...

yum -y install nano
nano /etc/selinux/config        selinix=disabled

systemctl disable firewalld
systemctl stop firewalld
systemctl status firewalld

yum install -y perl-DBD-MySQL

bash <( curl -s https://cdn.rawgit.com/sipcapture/homer-installer/master/homer_installer.sh )

anyway if i reboot the system  or restart mysqld service 
1. there is no way to login the UI as the user/password is not recognized..
2. homer rotate script doesen't work anymore

if i rerun the script again i am able to login the UI again but the rotate script don't want to work..

any suggestions?

Best Regards  
Marco

**Comment by lmangani** [comment]
@voipmarco could you please open your own separate issue? It seems unrelated to this thread.
At first sight it seems your DB is not persistent. Is this a normal host or some virtualization/container?
